fem/field_interpolant.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+// Implementation of Field Interpolants and necessary (Vector)QuadratorIntegrators
+
+#include "field_interpolant.hpp"
+#include "../linalg/densemat.hpp"
+#include "fem.hpp"
+
+namespace mfem
+{
+
+void VectorQuadratureIntegrator::AssembleRHSElementVect(const FiniteElement &fe,
+                                                        ElementTransformation &Tr,
+                                                        Vector &elvect)
+{
```

Comment:
This is the place to add an `MFEM_ASSERT` to check if the quadrature rule used in the integrator is the same as the `QuadratureFunction's` rule.